bdk_core/src/byte_list.rs

```diff
@@ -95,19 +99,20 @@ impl<F: Read + Write + Seek> DiskList<F> {
         header[0..8].copy_from_slice(tail_loc.to_be_bytes().as_ref());
         header[8..12].copy_from_slice((data.len() as u32).to_be_bytes().as_ref());
         // Write new entry first so we're sure it's written before we update pointer to tail
-        self.inner.write_all(&header[..])?;
+        self.inner.write_all(&header)?;
         self.inner.write_all(data)?;
         self.inner.flush()?;
```

Comment:
I did some digging and it turns out that this is wrong. We should be using `File` type here not `Read + Write + Seek`: https://doc.rust-lang.org/src/std/sys/unix/fs.rs.html#1061

Instead of `.flush` we should call `.datasync`.